warp-core/src/main/scala/com/workday/warp/arbiters/ArbiterLike.scala

```diff
@@ -25,6 +27,67 @@ trait ArbiterLike extends PersistenceAware {
   def vote[T: TestExecutionRowLikeType](ballot: Ballot, testExecution: T): Option[Throwable]
 
 
+  /**
+    * Wraps a vote with spike filtering.
+    *
+    * @param ballot box used to register vote result.
+    * @param testExecution [[TestExecutionRowLikeType]] we are voting on.
+    * @param spikeFilteringEnabled whether spike filtering is enabled.
+    * @param alertOnNth exceed limit.
+    * @return
+    */
+  final def voteWithSpikeFiltering[T: TestExecutionRowLikeType](ballot: Ballot,
+                                                                testExecution: T,
+                                                                spikeFilteringEnabled: Boolean,
+                                                                alertOnNth: Int): Option[Throwable] = {
```

Comment:
What happens if `alertOnNth` is negative?

Reply:
good point, we should enforce that this is strictly positive on the frontend side.